File: pwa/PwaClient/Models/Computer.cs
public class Computer
{
    public string Name { get; set; }
    public string Type { get; set; }
    public string EnclosureName { get; set; }
    public string CpuName { get; set; }
    public Ram Ram { get; set; }
    public int HardDiskCapacity { get; set; }
    public Gpu Gpu { get; set; }
    public int PowerSupply { get; set; }
    public int Price { get; set; }

    // Konstruktor
    public Computer(string name, string type, string enclosureName, string cpuName, Ram ram,
                    int hardDiskCapacity, Gpu gpu, int powerSupply, int price)
    {
        this.Name = name;
        this.Type = type;
        this.EnclosureName = enclosureName;
        this.CpuName = cpuName;
        this.Ram = ram;
        this.HardDiskCapacity = hardDiskCapacity;
        this.Gpu = gpu;
        this.PowerSupply = powerSupply;
        this.Price = price;
    }
}

File: mobile_client/lib/models/computer.dart
import 'dart:convert';

import 'package:mobile_client/models/gpu.dart';
import 'package:mobile_client/models/ram.dart';

List<Computer> computersFromJson(String str) => List<Computer>.from(json.decode(str).map((x) => Computer.fromJson(x)));

class Computer {
  final String name;
  final String type;
  final String enclosureName;
  final String cpuName;
  final Ram ram;
  final int hardDiskCapacity;
  final Gpu gpu;
  final int powerSupply;
  final int price;

  Computer({required this.name, required this.type, required this.enclosureName, required this.cpuName, required this.ram, required this.hardDiskCapacity, required this.gpu, required this.powerSupply, required this.price});

  factory Computer.fromJson(Map<String, dynamic> json) => Computer(
        name: json["name"],
        type: json["type"],
        enclosureName: json["enclosureName"],
        cpuName: json["cpuName"],
        ram: Ram.fromJson(json["ram"]),
        hardDiskCapacity: json["hardDiskCapacity"],
        gpu: Gpu.fromJson(json["gpu"]),
        powerSupply: json["powerSupply"],
        price: json["price"],
      );

  Map<String, dynamic> toJson() => {
        "name": name,
        "type": type,
        "enclosureName": enclosureName,
        "cpuName": cpuName,
        "ram": ram.toJson(),
        "hardDiskCapacity": hardDiskCapacity,
        "gpu": gpu.toJson(),
        "powerSupply": powerSupply,
        "price": price,
      };
}